### PR TITLE
Replace sprintf_s() and wsprintf() with StrFormat()

### PR DESCRIPTION
**Pull request by kiyolee**
This PR deals with all the `sprintf_s()` and `wsprintf()` calls and replaces with equivalent `StrFormat()`.
This leads to some possible improvements to SerialComms and `GetSymbol()`/`FormatAddress()` which I separate into individual commits to help review. (I can do if these deserve some separate PRs)
The remaining bigger lot of `sprintf()` will be deal with in another PR.

### CODE REVIEW

File: source/Debugger/Debug.cpp [hunk 1]
```diff
@@ -8420,33 +8411,33 @@ void DebugContinueStepping(const bool bCallerWillUpdateDisplay/*=false*/)
 
 		if (regs.pc == g_nDebugStepUntil || g_bDebugBreakpointHit)
 		{
-			char szStopMessage[CONSOLE_WIDTH];
-			const char* pszStopReason = szStopMessage;
+			std::string strStopMessage;
```
Comment (tomcw) [inline source/Debugger/Debug.cpp:8414]: Minor thing: you can get rid of `strStopMessage`

**Comment by tomcw** [comment]
I've looked in some detail at the non-debugger changes, and they appear OK.
I've skimmed the debugger changes - and again they appear OK.

If I were doing a thorough review then I should first checkout @kiyolee's branch, and check all code paths... but the changes are many and pervasive!

I think if there are some subtle issues, then they can just be fixed in the main branch (ie. post merge).
And it will be worth doing a Coverity static analysis scan, and see what new defects that reports.

So I will merge to master (then do the Coverity scan).

**Comment by tomcw** [comment]
NB. Coverity didn't report any new defects with any of the changes in this PR.

**Comment by tomcw** [comment]
btw, @kiyolee - thanks for your work on all of these string-formatting related changes.

**Review by Michaelangel007 — COMMENTED** [review]
Looks fine.

File: source/Debugger/Debugger_Help.cpp [hunk 2]
```diff
@@ -735,96 +733,78 @@ Update_t CmdHelpSpecific (int nArgs)
 //		if (nFound && (! bAllCommands) && (! bCategory))
 		if (nFound && (! bAllCommands) && bDisplayCategory)
 		{
-			char sCategory[ CONSOLE_WIDTH ];
+			const char* pszCategory = "";
```
Comment (Michaelangel007) [inline source/Debugger/Debugger_Help.cpp:736]: Please rename `pszCategory` to `psCategory`.  The `z` indicator is redundant.  All we need to know is that:

* This is a pointer
* This is a string

**Review by Michaelangel007 — COMMENTED** [review]
Other then the one var name, looks fine.

File: source/Debugger/Debugger_Help.cpp (hunk 2 above)
Comment (kiyolee) [reply source/Debugger/Debugger_Help.cpp:736]: https://github.com/AppleWin/AppleWin/pull/1057
Submitted the PR to fix this.

**Comment by Michaelangel007** [comment]
+1 Ship it!

**Pull request merged**